pandas/tests/test_groupby.py

```diff
@@ -1462,42 +1462,81 @@ def test_aggregate_api_consistency(self):
+                        'D': np.arange(8)})
 
         grouped = df.groupby(['A', 'B'])
-        result = grouped[['D', 'C']].agg({'r': np.sum, 'r2': np.mean})
-        expected = pd.concat([grouped[['D', 'C']].sum(),
-                              grouped[['D', 'C']].mean()],
-                             keys=['r', 'r2'],
-                             axis=1).stack(level=1)
-        compare(result, expected)
-
-        result = grouped[['D', 'C']].agg({'r': {'C': np.sum},
-                                          'r2': {'D': np.mean}})
-        expected = pd.concat([grouped[['C']].sum(),
-                              grouped[['D']].mean()],
+        c_mean = grouped['C'].mean()
+        c_sum = grouped['C'].sum()
+        d_mean = grouped['D'].mean()
+        d_sum = grouped['D'].sum()
+
+        result = grouped['D'].agg(['sum', 'mean'])
+        expected = pd.concat([d_sum, d_mean],
+                             axis=1)
+        expected.columns = ['sum', 'mean']
+        assert_frame_equal(result, expected, check_like=True)
+
+        result = grouped.agg([np.sum, np.mean])
+        expected = pd.concat([c_sum,
+                              c_mean,
+                              d_sum,
+                              d_mean],
                              axis=1)
-        expected.columns = MultiIndex.from_tuples([('r', 'C'), ('r2', 'D')])
-        compare(result, expected)
+        expected.columns = MultiIndex.from_product([['C', 'D'],
+                                                    ['sum', 'mean']])
+        assert_frame_equal(result, expected, check_like=True)
 
         result = grouped[['D', 'C']].agg([np.sum, np.mean])
-        expected = pd.concat([grouped['D'].sum(),
-                              grouped['D'].mean(),
-                              grouped['C'].sum(),
-                              grouped['C'].mean()],
+        expected = pd.concat([d_sum,
+                              d_mean,
+                              c_sum,
+                              c_mean],
                              axis=1)
-        expected.columns = MultiIndex.from_product([['D', 'C'], ['sum', 'mean']
-                                                    ])
-        compare(result, expected)
+        expected.columns = MultiIndex.from_product([['D', 'C'],
+                                                    ['sum', 'mean']])
+        assert_frame_equal(result, expected, check_like=True)
+
+        result = grouped[['D', 'C']].agg(OrderedDict([('r', np.sum),
+                                                      ('r2', np.mean)]))
+        expected = pd.concat([d_sum,
+                              c_sum,
+                              d_mean,
+                              c_mean],
+                             axis=1)
+        expected.columns = MultiIndex.from_product([['r', 'r2'],
+                                                    ['D', 'C']])
+        assert_frame_equal(result, expected, check_like=True)
```

Comment:
i believe I have all combos tested, but will validate in followup issue (which I am planning on checking boxes as much as possible on this PR though)